src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -853,16 +865,26 @@ bool async_manifest_view::in_archive(async_view_search_query_t o) {
                  && ko < _stm_manifest.get_start_kafka_offset().value_or(
                       kafka::offset::min());
       },
-      [this](model::timestamp ts) {
+      [this](async_view_timestamp_query ts_query) {
           // The condition for timequery is tricky. With offsets there is a
           // clear pivot point. The start_offset of the STM manifest separates
           // the STM region from the archive. With timestamps it's not as
           // simple.There could be a gap between the last segment in the archive
           // and the first segment in the STM manifest. We need in_stm and
           // in_archive to be consistent with each other. To do this we can use
           // last timestamp in the archive as a pivot point.
-          return _stm_manifest.get_spillover_map().last_segment()->max_timestamp
-                 >= ts;
+          bool range_overlaps
+            = ts_query.min_offset
+                < _stm_manifest.get_start_kafka_offset().value_or(
```

Comment:
Should this be <=?

Reply:
For this and the range_overlaps below, WDYT about writing the condition with the bounded offset interval? I guess as written we may have unbounded offsets, but perhaps the cases where there is a nullopt can be explicitly handled? It's otherwise a bit of gymnastics to parse what this is saying

Reply:
Updated in https://github.com/redpanda-data/redpanda/pull/18097/commits/75d91ee9c8cd2e8195dc03edd6d6308cc6a151bd.

It should be clearer now. < was correct here because I'm comparing it against the start of stm offset which implicitly acts as the end of the spillover. Made it more explicit now.

Reply:
bounded offset interval currently supports only model::offset but i'm using kafka::offests here. Leaving it for the future.